### PR TITLE
Docs: reorganization of panel plugin topics

### LINKED ISSUE #925: Reorganize plugin-tools site based on new design plan
Fix problems with less than ideal plugin-tools organization. Incorporate feedback received in Q1 from many stakeholders, external and internal to the plugins team and approved in a design plan.

This process includes the following:
- Move or rename folders and files
- Fix broken links on site caused by move/renames
- Fix broken links on other sites, principally grafana/grafana and plugin-examples
- Create new topic placeholders for new topics, make edits and/or assign to developers to make edits
- Test all changes through build process on Docker site and dev site
```[tasklist]
### Tasks
- [x] Reorganize 'reference' section
- [x] Reorganize 'introduction' section
- [ ] Reorganize 'create-a-plugin/develop-a-plugin' section
- [ ] Reorganize 'create-a-plugin/extend-a-plugin' section
```

### PR DESCRIPTION
**Pull request by josmperez**
Complete 4th stage of the plugin-tools site reorganization project. Move/rename site, fix broken links.

Note that there is only 1 redirect created in this PR despite moves of 3 files. The other 2 files will be redirected to the corresponding topic in "how-to-guides/data-source-plugins".

Fixes: https://github.com/grafana/plugin-tools/issues/925

Instructions for testing redirects: docusaurus/website/README.md#Redirects

Testing: Refer to [design doc](https://docs.google.com/document/d/1T3v41Ik5hOYNL0xMtPLOiK38IPGUlvsEgzFkDKvsROM/edit)

### CODE REVIEW

File: docusaurus/docs/how-to-guides/_category_.json
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/_category_.json:1]: you have this in #960

File: docusaurus/docs/how-to-guides/index.md
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/index.md:1]: again this is in #960

File: docusaurus/docs/how-to-guides/panel-plugins/index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+id: data-source-plugins
+title: Data source plugins
+description: Learn how to perform common tasks in Grafana data source plugin development.
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/panel-plugins/index.md:4]:
```suggestion
description: Learn how to add additional functionality and capabilities to Grafana panel plugins.
```

File: docusaurus/docs/how-to-guides/panel-plugins/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - develop
+  - development
+  - panel
+---
+
+Learn how to perform common tasks in Grafana panel plugin development.
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/panel-plugins/index.md:14]:
```suggestion
Learn how to add additional functionality and capabilities to Grafana panel plugins.
```

File: docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md [hunk 3]
```diff
@@ -25,42 +24,7 @@ If you want to migrate an existing plugin to use the data frame format, refer to
...
-```ts
-const series = [
-  { Time: 1599471973065, Value: 12.3 },
-  { Time: 1599471975729, Value: 28.6 },
-];
-
-const frame = toDataFrame(series);
-frame.name = 'http_requests_total';
-```
+If you build a data source plugin, then you'll most likely want to convert a response from an external API to a data frame. Learn how to do this in [Create a data frame](../../key-concepts/data-frames)
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md:27]: I don't think we'd want to keep this as a header?

**Review by sympatheticmoose — CHANGES_REQUESTED** [review]
This is getting messy as its overlapping significantly with #960 I'd strongly suggest against changing the same file (data frames) in both PRs

File: docusaurus/docs/how-to-guides/panel-plugins/error-handling-for-panel-plugins.md [hunk 4]
```diff
@@ -54,24 +54,6 @@ try {
 
 :::note
 
-Grafana displays the exception message in the UI as written, so use grammatically correct sentences. For more information, refer to the [Documentation style guide](https://grafana.com/docs/writers-toolkit/).
+Grafana displays the exception message in the UI as written, so use grammatically correct sentences. For more information, refer to the [documentation style guide](https://grafana.com/docs/writers-toolkit/).
 
 :::
-
```
Comment (academo) [inline docusaurus/docs/how-to-guides/panel-plugins/error-handling-for-panel-plugins.md:60]: this information looks valuable. why are we removing it?

File: docusaurus/docs/how-to-guides/_category_.json
Comment (josmperez) [reply docusaurus/docs/how-to-guides/_category_.json:1]: deleted duplicate data-frames.md and _category_.json files

File: docusaurus/docs/how-to-guides/index.md
Comment (josmperez) [reply docusaurus/docs/how-to-guides/index.md:1]: done

File: docusaurus/docs/how-to-guides/panel-plugins/index.md (hunk 1 above)
Comment (josmperez) [reply docusaurus/docs/how-to-guides/panel-plugins/index.md:4]: deleted duplicate. will alter description in other PR

File: docusaurus/docs/how-to-guides/panel-plugins/index.md (hunk 2 above)
Comment (josmperez) [reply docusaurus/docs/how-to-guides/panel-plugins/index.md]: done

File: docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md (hunk 3 above)
Comment (josmperez) [reply docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md:27]: i could argue either way about this header (it was intended to be a link to related info). deleting.

File: docusaurus/docs/how-to-guides/panel-plugins/error-handling-for-panel-plugins.md (hunk 4 above)
Comment (josmperez) [reply docusaurus/docs/how-to-guides/panel-plugins/error-handling-for-panel-plugins.md:60]: i believe the content is related to data source plugins, so it is included in a separate file in the data source reorg PR. i agree that by splitting the error handling topics into 2 separate entities, we are creating content that could probably benefit from improvement. let's do a separate PR later to improve these, and perhaps add some info back that's specifially relevvant to panel plugins

File: docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md (hunk 3 above)
Comment (josmperez) [reply docusaurus/docs/how-to-guides/panel-plugins/read-data-frames.md]: removed

**Comment by sympatheticmoose** [comment]
@josmperez I've quickly spun these changes into a new PR to avoid the issues that were occurring after different files were edited and then removed. PTAL when you have a chance 🙏 https://github.com/grafana/plugin-tools/pull/985

File: docusaurus/docs/how-to-guides/_category_.json
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/_category_.json:1]: note that deleting these files means that you're removing them from the site. Instead you need to undo any changes compared to whats in main.

File: docusaurus/docs/how-to-guides/index.md
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/index.md:1]: as above

**Comment by josmperez** [comment]
Having approved [985, ](https://github.com/grafana/plugin-tools/pull/985), I think we can close this now.

**Pull request closed without merging**